rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,571 @@
+
+Event type: `com.mbta.ctd.glides.imported-scheduled-trips`
+Fields in the event:
+- `scheduleId` (string)
+- `trips`: Array of:
+	- `route` (string)
+	- `startLocation` (Location)
+	- `departureTime` (Time)
+	- `endLocation` (Location)
+	- `arrivalTime` (Time)
+	- `runs` (Array of `runId`): the runs operating this trip, with the lead car in the consist first.
+
+*Notes*
+- the combination of `startLocation`, `endLocation`, `departureTime`, and `arrivalTime` must be unique for the `scheduleId`.
+
+### Operator signed in
+At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They do this by either physically signing a paper trainsheet, or by typing their badge number into Glides. This event represent either type of signin, but may not be generated in the case where an operator only signs into the paper trainsheet.
+
+Event type: `com.mbta.ctd.glides.operator-signed-in`
+Fields in the event:
+- `author` (Author): the inspector who signed in the operator
+- `operator` (Operator): the operator who signed in
+- `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
+
+### Trip added
+This creates a new trip in the timesheet. It may or may not be a new trip relative to GTFS (see the splitting a train reference example).
+
+Event type: `com.mbta.ctd.glides.trip-added`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): provides a unique identifier for the newly created trip
+- `route` (string, optional): if specified, the route tag for the trip. This is different from a GTFS route ID.
+- `startLocation` (Location, conditionally required): where the trip will be starting. Required if `departureTime` is specified.
+- `endLocation` (Location, conditionally required): where the trip will be ending. Required if `arrivalTime` is specified.
+- `departureTime` (Time, conditionally required): when the added trip is expected to depart `startLocation`.
+- `arrivalTime` (Time, conditionally required): when the added trip is expected to arrive at `endLocation`.
+- `previousTripKey` (Trip Key, optional): links the newly created trip to the trip immediately before it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message. Required if `endLocation` is specified and `arrivalTime` is NOT specified.
+- `nextTripKey` (Trip Key, optional): links the newly created trip to the trip after it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message. Required if `startLocation` is specified and `departureTime` is NOT specified.
+- `consist` (array of Optional Car, optional, default: empty list): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist.
```

Comment:
A piece of information that isn't explicitly captured here (and in `trip-updated`) is the presence or absence of the second car. When adding a trip, we can use an array of length 1 to denote a one-car train and an array of length 2 to denote a two-car train; if we do, those semantics should be defined here.